swarm/runtime/safe_paths.py

```
@@ -46,3 +47,35 @@ def validate_path_component(component: str, name: str = "path component") -> str
+        The validated path string.
+
+    Raises:
+        ValueError: If the path is invalid.
+    """
+    if not path:
+        raise ValueError(f"{name} cannot be empty")
+
+    p = Path(path)
+
+    if p.is_absolute():
+        raise ValueError(f"{name} must be a relative path")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)